commands/operator-sdk/cmd/add/api.go

```diff
@@ -146,7 +149,13 @@ func updateRoleForResource(r *scaffold.Resource, absProjectPath string) error {
 			role.Rules = append(role.Rules, *pr)
 		}
 		// update role.yaml
-		data, err := yaml.Marshal(&role)
+		d, err := json.Marshal(&role)
```

Comment:
Was `yaml.Marshal` not returning the expected yaml for a role? Why is this change necessary?

Reply:
Yeah, that was not working at all. 

yaml.Marshal was not outputting the correct yaml, this is because there are no tags for yaml in the role struct, meaning 1. names were incorrect and 2. that type meta was being printed.